dvc/repo/checkout.py

```diff
@@ -100,8 +101,13 @@ def checkout(
             for key, items in result.items():
                 stats[key].extend(_fspath_dir(path) for path in items)
 
-    if stats.get("failed"):
-        raise CheckoutError(stats["failed"], stats)
+    if stats.get("failed") or stats.get("failed_not_stored"):
+        raise CheckoutError(
+            stats["failed"] + stats["failed_not_stored"],
+            stats,
+            failed_not_stored=stats["failed_not_stored"],
```

Comment:
Can we extract it inside CheckoutError from `stats`?

Reply:
Yes we could do that, but we'd still have to keep the `failed_not_stored` entry on l.61 above. Unless we made it a `defaultdict` like in 
https://github.com/iterative/dvc/blob/68471ebbf073b7c893ba3f9af17a09dc9425d761/dvc/stage/__init__.py#L450-L457

That way `checkout` doesn't need to know about the structure of stats except for how to check if there are errors in 
https://github.com/iterative/dvc/blob/68471ebbf073b7c893ba3f9af17a09dc9425d761/dvc/repo/checkout.py#L104-L105

Alternatively, since stats is becoming complex, it could be made into an object with a `.failed` property which `checkout` could test. Then `CheckoutError` could extract the details without `checkout` needing to know about them.